File: gap/bar.pl
#!/usr/bin/env perl

print("stamp\tnow\tgap\n");
foreach $line (`cat d`) {
    ($stamp, $data) = split(/\s+/, $line);
    
    $now = $data;
    $gap = $now - $post;

    printf("$stamp\t$now\t$gap\n");
    
    $post = $now;
}

File: gap/bar.py
#!/usr/bin/env python

with open('d3') as f:
    data = f.readlines()

post = 0
for line in data:
    stamp, value = line.split()
    now = int(value.replace(',',''))
    gap = now - post
    print("{} {:,} {:,}".format(stamp, now, gap))
    post = now
